src/math/CMakeLists.txt

```diff
@@ -5,31 +5,41 @@ if(BUILD_LIBRARY)
 	return()
 endif()
 
+is_zephyr(it_is_zephyr)
+
 add_local_sources(sof numbers.c)
 
-if(CONFIG_CORDIC_FIXED)
+if(CONFIG_CORDIC_FIXED OR it_is_zephyr)
         add_local_sources(sof trig.c)
 endif()
 
 add_local_sources_ifdef(CONFIG_SQRT_FIXED sof sqrt_int16.c)
 
 add_local_sources_ifdef(CONFIG_MATH_EXP sof exp_fcn.c exp_fcn_hifi.c)
 
-if(CONFIG_MATH_DECIBELS)
+if(CONFIG_MATH_DECIBELS OR it_is_zephyr)
         add_local_sources(sof decibels.c)
 endif()
 
+# Never used with Zephyr.
+if(NOT it_is_zephyr)
 add_local_sources_ifdef(CONFIG_NATURAL_LOGARITHM_FIXED sof log_e.c)
 
 add_local_sources_ifdef(CONFIG_COMMON_LOGARITHM_FIXED sof log_10.c)
 
 add_local_sources_ifdef(CONFIG_POWER_FIXED sof power.c)
 
 add_local_sources_ifdef(CONFIG_BINARY_LOGARITHM_FIXED sof base2log.c)
+endif()
 
+if(it_is_zephyr)
```

Comment:
Same here!! Using the same name for managing math functions under Zephyr is not a good idea.